DESIGN.md

```diff
@@ -0,0 +1,168 @@
+    ...
+    ...
+
+    private Network createVnetAndSubnets(
+            FlightContext context,
+            ArmManagers armManagers,
+            String vNetName,
+            String networkSecurityGroupId) {
+      var landingZoneId =
+              getParameterOrThrow(
+                      context.getInputParameters(), LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+
+      try {
+        return armManagers
+                .azureResourceManager()
+                .networks()
+                .define(vNetName)
+                .withRegion(getMRGRegionName(context))
+                .with()
+                ...
+                .create()
+      ...
+        
+    @Override
+    public String getResourceType() {
+      return "VirtualNetwork";
+    }
+
+    ...
+        
+    @Override
+    public List<ResourceNameRequirements> getResourceNameRequirements() {
+      return List.of(
+              new ResourceNameRequirements(
+                      getResourceType(), ResourceNameGenerator.MAX_VNET_NAME_LENGTH));
+    }
+```
+
+It is important that step's implementation should be idempotent. Please take a look at Stairway developer guide [here](https://github.com/DataBiosphere/stairway/blob/develop/FLIGHT_DEVELOPER_GUIDE.md).
+
+Resource creation is defined using the standard Azure Java SDK. Together with defining desired properties for a resource it is also important to assign specific tags to a resource.
+
+### Receiving Parameters
+
+Each step has access to [ParameterResolver](https://github.com/DataBiosphere/terra-landing-zone-service/blob/main/library/src/main/java/bio/terra/landingzone/library/landingzones/definition/factories/ParametersResolver.java). This class allows accessing specific parameters. All default values for parameters are set in `LandingZoneDefaultParameters`.
+
+### Naming Resources and Idempotency
+
+Each step which creates an Azure resource should provide requirements for name generation. For doing this, it should provide a unique resource type (unique across all steps within the flight)
+and also return a list of `ResourceNameRequirements` (in general, a step can create more than one Azure resource). Please take a look at the step example above.
+
+The resource name generator creates a name from a hash of the landing zone id and internal sequence number.
+As long as the landing zone id is globally unique, the resulting name will be the same across retry attempts with a very
+low probability of a naming collision.
+
+### Error handling and retrying
+
+Each step is responsible for error handling. In the case that a step is responsible for creation of just one resource (recommended) there is usually no need to add additional error handling,
+but this can be changed depending on complexity/requirements. The base step from which all steps are inherited contains some common error handling as well;
+for instance, the base class handles the situation when the resource already exists.
+
+## Deleting Landing Zones
```

Comment:
The deletion part is taken from a standalone `DELETING_RESOURCES.md` file that was the only thing in a docs directory. I have no idea how accurate this information is, as it hasn't been updated since it was initially written by Jesus 2 years ago.

Reply:
Ahhhh ... this is making some sense re: copy-and-paste (so, the inline code may have already been in the markdown documents) - is this conclusion right?

Reply:
Yes, I mostly just moved things around, and certainly didn't add any inline code.

Reply:
I wouldn't sweat it now then if you don't want to ... just a possible "TODO" for simplification for later - 🤓